builder/azure/common/vault.go

```diff
@@ -137,7 +137,7 @@ func (client *VaultClient) DeletePreparer(resourceGroupName string, vaultName st
 func (client *VaultClient) DeleteSender(req *http.Request) (*http.Response, error) {
 	return autorest.SendWithSender(client,
 		req,
-		azure.DoPollForAsynchronous(client.PollingDelay))
+		azure.DoRetryWithRegistration(client.Client))
```

Comment:
This is not exactly the same, but fine, because it's the delete call and I don't expect deleting a KV to take a long time anyway.